dash/development/_py_components_generation.py

```diff
@@ -164,7 +165,7 @@ def generate_class_file(typename, props, description, namespace):
     print('Generated {}'.format(file_name))
 
 
-def generate_imports(project_shortname, components):
+def generate_imports(project_shortname, components, metadata=None):
```

Comment:
I think R will need the package.json info too, we should also pass that argument too.

Reply:
Can we add that in at the same time as the R stuff instead?

Reply:
Actually I think it's just a confusion, `metadata` should be `package_info`. We already mapped the `metadata.json` in generate_component_files, it's the `package.json` mapping that is needed for R. So just rename the argument to `package_info`.

Reply:
That doesn’t make sense to me. The file dump line I moved writes to metadata.json

Reply:
Is it wrong on master now?

Reply:
No, I forgot about the move of the metadata write, let's keep this as is and we'll modify the functions as needed in the R generation PR.